### ISSUE TITLE
CLI as JAR to SDKMan instead of Native for 3.x

### ISSUE THREAD

**Issue by sdelamo** [micronaut-projects/micronaut-starter#2072]
see: https://github.com/micronaut-projects/micronaut-starter/actions/workflows/mn-macos-snapshot.yml?query=branch%3A3.8.x

```
Adding graalvm-ce-java11-darwin 21.3.0 to tool-cache ...
/Users/runner/hostedtoolcache/graalvm-ce-java11-darwin/21.3.0/x64/graalvm-ce-java11-21.3.0/Contents/Home/bin/gu --non-interactive install --no-progress native-image
Downloading: Release index file from oca.opensource.oracle.com
Error: Error reading component list: https://oca.opensource.oracle.com/gds/meta-data.json
```

**Comment by timyates** [comment]
So I think instead of using `native-image` as we do in the 3.8.x actions

https://github.com/micronaut-projects/micronaut-starter/blob/3.8.x/.github/workflows/mn-linux-snapshot.yml#L30

We need to use nativeCompile as we do in the 4.1.x actions

https://github.com/micronaut-projects/micronaut-starter/blob/4.1.x/.github/workflows/mn-linux-snapshot.yml#L26

**Comment by timyates** [comment]
@sdelamo I think this is attached to the wrong project?

I've raised PRs to fix native compilation for [3.8.x](https://github.com/micronaut-projects/micronaut-starter/pull/2073), [3.9.x](https://github.com/micronaut-projects/micronaut-starter/pull/2075) and [3.10.x](https://github.com/micronaut-projects/micronaut-starter/pull/2076)
